packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -256,30 +256,34 @@ class IOSDevice extends Device {
       // ports post launch.
       printTrace("Debugging is enabled, connecting to observatory and the diagnostic server");
 
-      Future<int> forwardObsPort = _acquireAndForwardPort(app,
-                                                          ProtocolDiscovery.kObservatoryService,
-                                                          debuggingOptions.observatoryPort);
-      Future<int> forwardDiagPort;
+      Future<Uri> forwardObsUri = _acquireAndForwardUriToNewPort(
+        app,
+        ProtocolDiscovery.kObservatoryService,
+        debuggingOptions.observatoryPort,
+      );
+      Future<Uri> forwardDiagUri;
```

Comment:
Agreed but out of scope for this PR. Added a TODO and I'll try and tackle that soon.